cmd/traffic/cmd/manager/internal/mutator/watcher.go

```diff
@@ -431,23 +497,59 @@ func (c *configWatcher) Store(ctx context.Context, ac *agentconfig.Sidecar, upda
 			},
 			ObjectMeta: meta.ObjectMeta{
 				Name:      agentconfig.ConfigMap,
-				Namespace: ns,
-			},
-			Data: map[string]string{
-				ac.AgentName: yml,
+				Namespace: c.namespace,
 			},
 		}
-		dlog.Debugf(ctx, "Creating new ConfigMap %s.%s with %s", agentconfig.ConfigMap, ns, ac.AgentName)
-		_, err = api.Create(ctx, cm, meta.CreateOptions{})
-	} else {
-		if _, ok := cm.Data[ac.AgentName]; ok {
-			dlog.Debugf(ctx, "Updating %s in ConfigMap %s.%s", ac.AgentName, agentconfig.ConfigMap, ns)
-		} else {
-			dlog.Debugf(ctx, "Adding %s to ConfigMap %s.%s", ac.AgentName, agentconfig.ConfigMap, ns)
+	}
+
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
+
+	cmData := map[string]string{}
```

Comment:
```suggestion
	var cmData map[string]string
```
Nit for sure, but I think it brings clarity. Initialization is unnecessary. If there is a `c.cw.data[c.namespace]`, then `cmData` is assigned to that value, otherwise a `nil`, value is OK.